content/en/docs/contributing/building.md

```diff
@@ -0,0 +1,121 @@
+* `:helm` is the part of the Bazel file to execute, these are defined in the Bazel config themselves.
+
+> **TIP**: `...` is a recursive lookup in Bazel, it will run al tests in all subfolders when set, it is also the easiest way to invoke them.
+> For example `bazel test //pkg/...` will test all tests in all packages.
+
+## Help so much Bazel!
+
+No worries we got a lot of helper scripts for you!
+Need to set up a local cluster and install cert-manager in it? Take a look at [our kind documentation](../kind/).
+
+We got a few very handy tools inside `./hack` and `./devel`. These are the most common ones which you can use:
+
+### Just update everything you can!
+
+Bazel takes care of a lot of automatic code generation for us, from generating CRD updates to updating it's own `BUILD.bazel` files.
+If you just want to do everything at once (and have 5 minutes of your time) you can run:
+
+```bash 
+$ ./hack/update-all.sh
+```
+
+This will update everything you need without having to care about what needs changing.
+
+> **NOTE:** we strongly reccomend to run this before you create a PR!
+
+### I need granular control 
+
+You can also pick and mix the individual bash helper scripts:
+* `update-bazel.sh`: updates the all `*.bazel` files includding formatting them
+* `update-codegen.sh`: runs all code generation
+* `update-crds.sh`: updates all CRD files to the lastest scheme
+* `update-deps-licenses.sh`: updates the `LICENCES` file, needed when adding/updating dependancies
+* `update-deps.sh`: installs new dependancies declared in the code and adds them into the Bazel and Go module files
+* `update-gofmt.sh`: runs `go fmt` on all code
+
+Most of these have a `verify-*` equivalent which will run inside our CI to verify all the scripts ran before merging the PR.
+
+## Building the project
+
+You can ask Bazel to build the code for you to run in your local machine.
+The output will end up in `bazel-out` on your disk.
+
+You can get the exact path by looking at the Bazel output:
+```
+Target //cmd/ctl:ctl up-to-date:
+  bazel-out/k8-fastbuild-ST-4c64f0b3d5c7/bin/cmd/ctl/kubectl-cert_manager
+```
+
+### Building the Go binaries (for local OS)
+
+You can build the controllers to run them locally using:
+```bash
+$ bazel build //cmd/...
+```
+If you need them inside a local cluster check out [our kind documentation](../kind/).
+
+### Building the CLI
+
+You can use `go run ./cmd/ctl` to quickly run the CLI.
+You can also compile it using Bazel:
+```bash
+$ bazel build //cmd/ctl
+```
+
+### Building images
+
+If you need the Docker images you can generate these using:
+```bash
+$ export APP_VERSION="dev"
+$ export DOCKER_REGISTRY="quay.io/jetstack/"
+$ bazel run \
+		--stamp \
+		--platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 \
+		//build:server-images
+```
+
+`--stamp` enabled reproducable builds while `--platforms` defines which images to build, in this example for AMD64 Linux.
```

Comment:
```suggestion
`--stamp` enables reproducible builds while `--platforms` defines which images to build, in this example for AMD64 Linux.
```

Reply:
wouldn't you add those next to the options (as comments) in the bash snippet itself? makes it more obvious to find which option means what imho